File: app/src/main/java/com/ladsoft/bakingapp/mvp/presenter/state/RecipeState.java
package com.ladsoft.bakingapp.mvp.presenter.state;

import android.os.Bundle;

import com.ladsoft.bakingapp.manager.SessionManager;
import com.ladsoft.bakingapp.mvp.RecipeMvp;

import java.io.Serializable;
import java.util.HashMap;

public class RecipeState implements RecipeMvp.StateContainer {
    private final SessionManager sessionManager;
    private long recipeId;

    public RecipeState(SessionManager sessionManager) {
        this.sessionManager = sessionManager;
    }

    public void load(Bundle bundle) {
        recipeId = bundle.getLong(RecipeMvp.StateContainer.EXTRA_RECIPE_ID, 0L);
        sessionManager.setLastSelectedReceiptId(recipeId);
    }

    public void load(Serializable serializable) {
        HashMap<String, Object> stateMap = (HashMap<String, Object>) serializable;

        recipeId = stateMap.containsKey(RecipeMvp.StateContainer.EXTRA_RECIPE_ID)
                                        ? (long) stateMap.get(RecipeMvp.StateContainer.EXTRA_RECIPE_ID)
                                        : 0L;
    }

    @Override
    public long getRecipeId() {
        return recipeId;
    }

    @Override
    public HashMap<String, Object> getStateMap() {
        HashMap<String, Object> stateMap = new HashMap<>();
        stateMap.put(RecipeMvp.StateContainer.EXTRA_RECIPE_ID, recipeId);

        return stateMap;
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/mvp/presenter/state/RecipeState.kt
package com.ladsoft.bakingapp.mvp.presenter.state

import android.os.Bundle
import com.ladsoft.bakingapp.manager.SessionManager
import com.ladsoft.bakingapp.mvp.RecipeMvp
import java.io.Serializable


class RecipeState: RecipeMvp.StateContainer {
    override var recipeId : Long = 0L
    val sessionManager: SessionManager

    constructor(sessionManager: SessionManager) {
        this.sessionManager = sessionManager
    }

    fun load(bundle: Bundle?) {
        (bundle?.getLong(RecipeMvp.StateContainer.EXTRA_RECIPE_ID) ?: 0L).apply {
            recipeId = this
            sessionManager.setLastSelectedReceiptId(this)
        }
    }

    fun load(serializable: Serializable?) {
        serializable?.apply {
            val stateMap = serializable as HashMap<*, *>
            recipeId = stateMap[RecipeMvp.StateContainer.EXTRA_RECIPE_ID] as Long
        }
    }

    override fun getStateMap(): HashMap<String, Any> {
        val stateMap = HashMap<String, Any>()
        stateMap.put(RecipeMvp.StateContainer.EXTRA_RECIPE_ID, recipeId)

        return stateMap
    }

}
